tests/infrastructure/virt-controller-leader-election.go

```diff
@@ -62,12 +53,9 @@ var _ = DescribeInfra("Start a VirtualMachineInstance", func() {
 			Eventually(libinfra.GetLeader, 30*time.Second, 5*time.Second).ShouldNot(Equal(leaderPodName))
 
 			By("Starting a new VirtualMachineInstance")
-			vmi := libvmifact.NewAlpine()
-			obj, err := virtClient.RestClient().Post().Resource("virtualmachineinstances").Namespace(testsuite.GetTestNamespace(vmi)).Body(vmi).Do(context.Background()).Get()
+			vmi, err := virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(nil)).Create(context.Background(), libvmifact.NewGuestless(), metav1.CreateOptions{})
 			Expect(err).ToNot(HaveOccurred())
-			vmiObj, ok := obj.(*v1.VirtualMachineInstance)
-			Expect(ok).To(BeTrue(), "Object is not of type *v1.VirtualMachineInstance")
-			libwait.WaitForSuccessfulVMIStart(vmiObj)
+			libwait.WaitForSuccessfulVMIStart(vmi)
```

Comment:
This is probably overkill, for verifying the controller we only need to check for `Scheduled`

Reply:
Unfortunately, `libwait` is using polling instead of a watch. Since `Scheduled` is a transient case, waiting for it would be raceful.

Can I do something else to gain your approval?

Reply:
We should support waiting for multiple phases, something like `Scheduled, Running`

Reply:
Actually we can go as far as `Scheduling` imho or `Pending`

Reply:
> Unfortunately, `libwait` is using polling instead of a watch. Since `Scheduled` is a transient case, waiting for it would be raceful.
> 

I have to take this statement back. `libwait` /does/ use watchers, so it is fine to use it with a transient state like `Pending`. I think this is a very minor improvement of the test, but reviewer is king.

Reply:
I don't remember us using watch.

Reply:
> I don't remember us using watch.

This worries me, as I see https://github.com/kubevirt/kubevirt/blob/6ef3888/tests/libwait/wait.go#L144 

If tests don't use watch, how can they wait for a transient state such as `Pending` ?